contracts/exchange-forwarder/contracts/src/MixinExchangeWrapper.sol

```diff
@@ -92,45 +92,90 @@ contract MixinExchangeWrapper is
         bytes[] memory signatures
     )
         internal
-        returns (FillResults memory totalFillResults)
+        returns (
+            FillResults memory totalFillResults,
+            uint256 wethSpentAmount,
+            uint256 makerAssetAcquiredAmount
+        )
     {
-        bytes memory makerAssetData = orders[0].makerAssetData;
-        bytes memory wethAssetData = WETH_ASSET_DATA;
-
         uint256 ordersLength = orders.length;
-        for (uint256 i = 0; i != ordersLength; i++) {
-
-            // We assume that asset being bought by taker is the same for each order.
-            // We assume that asset being sold by taker is WETH for each order.
-            orders[i].makerAssetData = makerAssetData;
-            orders[i].takerAssetData = wethAssetData;
 
-            // Calculate the remaining amount of WETH to sell
-            uint256 remainingTakerAssetFillAmount = _safeSub(wethSellAmount, totalFillResults.takerAssetFilledAmount);
+        for (uint256 i = 0; i != ordersLength; i++) {
+            require(
```

Comment:
We should make this a RichError instead of a normal revert. Let me know if you have questions regarding how to use RichErrors.